File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {

        int total_product = 1;
        int zero_count = 0;

        for (int n : nums) {
            if (n == 0) {
                zero_count++;
            } else {
                total_product *= n;
            }
        }

        vector<int> sol;

        for (int n : nums) {

            if (zero_count > 1) {
                sol.push_back(0);
            }
            else if (zero_count == 1) {
                sol.push_back(n == 0 ? total_product : 0);
            }
            else {
                sol.push_back(total_product / n);
            }
        }

        return sol;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        for (int n : nums) {
            int count = freq.getOrDefault(n, 0);
            freq.put(n, count + 1);
        }
        PriorityQueue<Map.Entry<Integer, Integer>> pq = new PriorityQueue<>((a, b) -> b.getValue() - a.getValue());
        pq.addAll(freq.entrySet());

        int[] sol = new int[k];
        for (int i = 0 ; i < k ; i++) {
            sol[i] = pq.poll().getKey();
        }
        return sol;
    }
}
